service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -131,9 +144,67 @@ func (r *Resource) getObject(ctx context.Context, v interface{}) (*vpa_types.Ver
 		},
 	}
 
+	mcObservabilityBundleAppVersion, err := r.getManagementClusterObservabilityBundleAppVersion(ctx, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	version, err := semver.Parse(mcObservabilityBundleAppVersion)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	if version.LT(semver.MustParse("1.2.0")) {
+		// Set target reference to Prometheus StatefulSet
+		vpa.Spec.TargetRef = &autoscaling.CrossVersionObjectReference{
+			Kind:       "StatefulSet",
+			Name:       key.PrometheusSTSName(cluster),
+			APIVersion: "apps/v1",
+		}
+	} else {
+		// Set target reference to Prometheus CR
+		vpa.Spec.TargetRef = &autoscaling.CrossVersionObjectReference{
+			Kind:       "Prometheus",
+			Name:       key.ClusterID(cluster),
+			APIVersion: "monitoring.coreos.com/v1",
+		}
+
+	}
+
 	return vpa, nil
 }
 
+const unknownObservabilityBundleVersion = "0.0.0"
```

Comment:
Is it good practice to declare a `const` in the middle of the file? Shouldn't it be in the beginning of the file?

Reply:
I would Say no 😂